### DISCUSSION TITLE
Why is the counter page, an auto rendermode page,  being served from the web server all the time?

### DISCUSSION THREAD

**Discussion by Tony20221** [dotnet/aspnetcore#52362] [Q&A]
I created a new Blazor web app using the project template in Visual Studio and .NET 8.0. I noticed the counter page which is in the client project, keeps being retrieved fully from the web server even though it has the '@rendermode InteractiveAuto' attribute. This is evidenced in devtools Network tab and using a network sniffer. My understanding of the auto render mode is that a page is retrieved from the web server the first time using Blazor server and when the browser gets all the wasm files and assemblies, subsequent requests for any auto render page will be served within the WASM app in the browser. 

Why is the counter page always being served from the server? I don't expect the counter page request to show up anymore in the network tab after the first request, What am I not understanding correctly?

**Comment by sultanAalyami** [comment]
See file program.cs Note that the counter is added to Assembly Or something like that, I think from here the server reference is set.
